sdk/core/Azure.Core/src/HttpMessage.cs

```diff
@@ -23,24 +24,26 @@ public sealed class HttpMessage : IDisposable
         /// <param name="request">The request.</param>
         /// <param name="responseClassifier">The response classifier.</param>
         public HttpMessage(Request request, ResponseClassifier responseClassifier)
+            : base(ToPipelineRequest(request))
         {
-            Argument.AssertNotNull(request, nameof(Request));
+            Argument.AssertNotNull(request, nameof(request));
+
+            _propertyBag = new ArrayBackedPropertyBag<ulong, object>();
+
             Request = request;
             ResponseClassifier = responseClassifier;
-            BufferResponse = true;
-            _propertyBag = new ArrayBackedPropertyBag<ulong, object>();
         }
 
         /// <summary>
         /// Gets the <see cref="Request"/> associated with this message.
         /// </summary>
-        public Request Request { get; }
+        public new Request Request { get; }
 
         /// <summary>
         /// Gets the <see cref="Response"/> associated with this message. Throws an exception if it wasn't set yet.
         /// To avoid the exception use <see cref="HasResponse"/> property to check.
         /// </summary>
-        public Response Response
+        public new Response Response
```

Comment:
Do we need to keep this here as opposed to just keeping the virtual on the base?

Reply:
I don't know how likely this is going to hit but it does have the potential for breaking folks. Right now, I'd say it is just a caution to be on the lookout for.

Reply:
Can you help me understand a little better how it could introduce a break?  If we decide to keep it, I'd like to make sure we have good test/CI coverage to prevent this outcome if we can, and understanding better would also help me think through possible mitigations if we go that route instead.

Reply:
It has been a while since I've looked at the details of these calls but IIRC the compiler might still generate a "callvirt" for folks that are trying to call this property via `base.Repsonse`. However thinking a little more this might only be a break if the new property didn't already exist in the middle. I tried to create a simple repro but wasn't able to for your case. I'll think about it a little more but for now just be caution with introducing new/virtual/overrides when introducing new base types where a derived type maybe calling base.<name>.

Reply:
Sounds good - thank you so much, @weshaggard!  If needed, we can reach out to folks on the compiler team to understand the risks here better.  Let me know if it's worth going that route.

Reply:
Here are some cases of problems that one can get by using `new`: https://dotnetfiddle.net/UnVOfp
Those are very simple - in real-life app there will be myriads of problems that our users will break their heads on.
The issue with 'new' modifier is that it is much more clear for runtime than for the user.
The only "safe" case of using 'new' modifier is to provide more specific version of the return type, e.g.:
```cs
public class C1 
{
    public M1 GetSomething() 
    {
        // do something
        var m1 = GetM1Instance();
        // do something more
        return m1;
    }
    protected virtual M1 GetM1Instance() => new M1();
}

public class C2 : C1
{
    public new M2 GetSomething() => (M2)base.GetSomething();
    protected override M1 GetM1Instance() => new M2();
}

public class M1{}
public class M2 : M1{}
```